### PR TITLE
imdsclient: update readme to reflect new helper functions

### PR DESCRIPTION
**Pull request by jpculp**
**Issue number:**

N/A

**Description of changes:**

This is an update to documentation as `fetch_dynamic` and `fetch_metadata` were previously replaced with more targeted functions.

**Testing done:**

Built `imdsclient` crate.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/imdsclient/README.md [hunk 1]
```diff
@@ -3,9 +3,19 @@
...
+- `fetch_instance_type`
+- `fetch_local_ipv4_address`
+- `fetch_mac_addresses`
+- `fetch_public_ssh_keys`
+- `fetch_region`
+- `fetch_userdata`
+
+For custom IMDS end targets, you can use `fetch_bytes` and `fetch_string`.
+
+To query IMDS without high-level wrappers, there is also a `fetch_imds` method.
```
Comment (zmrow) [inline sources/imdsclient/README.md]: Nit: Just from reading this, I'm not sure when I would use `fetch_bytes/string` vs. `fetch_imds`.  Can we clarify a bit?

**Comment by jpculp** [comment]
Updated for further clarity.

**Review by zmrow — APPROVED** [review]
:ice_cream:

**Review by webern — COMMENTED** [review]
I actually think it would be better to remove the enumeration of high level function names since it will always be in danger of being incorrect.

Also the blurbs about fetch_imds, fetch_bytes and fetch_string are incorrect. These are private functions and as-such are implementation details that shouldn't go in the readme.

**Comment by jpculp** [comment]
Hm. I forgot that we switched the last 3 to private, but now that I think about it I wonder if that was really the right thing to do.

**Comment by jpculp** [comment]
Took a different approach to the documentation.

**Review by webern — APPROVED** [review]
I like this version much better!

**Review by zmrow — APPROVED** [review]

**Comment by jpculp** [comment]
Changed ``` to ` so cargo doesn't try to test the example uri.

**Review by zmrow — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Pull request merged**